### PR TITLE
Port some safe bits to c++

### PR DESCRIPTION
**Pull request by adamwight**
Just some minor patches which update the code inside of modules.  Changes on this branch are intended to not ripple over to dependencies outside of each file.

### CODE REVIEW

File: gnucash/gnucash.cpp [hunk 1]
```diff
@@ -195,9 +195,9 @@ scm_run_gnucash (void *data, [[maybe_unused]] int argc, [[maybe_unused]] char **
 
     gnc_hook_run(HOOK_STARTUP, NULL);
 
-    boost::optional<boost::filesystem::path> fn;
     if (!user_file_spec->nofile)
     {
+        boost::optional<boost::filesystem::path> fn;
         if (auto fn = get_file_to_load (user_file_spec->file_to_load))
```
Comment (gjanssens) [inline gnucash/gnucash.cpp:200]: You redefine fn inside the `if` clause in the next line. I think you can drop this declaration.

Comment (adamwight) [reply gnucash/gnucash.cpp:200]: Thanks, pushing a correction now!

File: gnucash/gnucash.cpp [hunk 2]
```diff
@@ -114,14 +116,14 @@ load_gnucash_modules()
...
 {
-    if (file_to_load && *file_to_load != '\0')
-        return g_strdup(file_to_load);
-    else
-        /* Note history will always return a valid (possibly empty) string */
-        return gnc_history_get_last();
+    if (!file_to_load.empty())
+        return file_to_load;
+    if (auto last = gnc_history_get_last())
+        return boost::filesystem::path(last);
```
Comment (jralls) [inline gnucash/gnucash.cpp:125]: Leaks `last`.

**Comment by jralls** [comment]
The boost::filename commit adds a great deal of overhead for no discernible purpose. The `user data_migration_msg` commit would add substantial overhead to no purpose except that `userdata_migration_msg` got moved to gnucash-core-app.cpp and this little fragment was left behind.  `userdata_migration_msg` is never changed from `nullptr` in gnucash.cpp, so the fragment should simply be removed.

**Pull request closed without merging**